src/libcore/ops/drop.rs

```diff
@@ -1,85 +1,141 @@
-/// Used to run some code when a value goes out of scope.
-/// This is sometimes called a 'destructor'.
+/// Custom code within the destructor.
 ///
-/// When a value goes out of scope, it will have its `drop` method called if
-/// its type implements `Drop`. Then, any fields the value contains will also
-/// be dropped recursively.
+/// When a value is no longer needed, Rust will run a "destructor" on that value.
+/// The most common way that a value is no longer needed is when it goes out of
+/// scope. Destructors may still run in other circumstances, but we're going to
+/// focus on scope for the examples here. To learn about some of those other cases,
+/// please see [the reference] section on destructors.
 ///
-/// Because of this recursive dropping, you do not need to implement this trait
-/// unless your type needs its own destructor logic.
+/// [the reference]: https://doc.rust-lang.org/reference/destructors.html
 ///
-/// Refer to [the chapter on `Drop` in *The Rust Programming Language*][book]
-/// for some more elaboration.
+/// This destructor consists of two components:
+/// - A call to `Drop::drop` for that value, in case `Drop` is implemented for its type.
```

Comment:
```suggestion
/// - A call to `Drop::drop` for that value, in case the special `Drop` trait is implemented for its type.
```
This is the first time `Drop` is mentioned, might be a good idea to introduce it as a "special trait"?

Reply:
I don't like `special` here. I feel like

> I am not sure what this slightly ominous announcement is telling me. It doesn't really help to understand the following text, does it?

also applies here.

Reply:
Hm. I feel like the text here already says why it is special -- because it is called by the destructor.

Reply:
Hmm, it feet like your wording may imply that there's some other `Drop` trait which is `special`.
Changed "the" to "this".

> A call to `Drop::drop` for that value, if this special `Drop` trait is implemented for its type.

Reply:
To me "this special drop trait" implies that there's another one ("this X and that X", making it multiple X), but "*the* special drop trait" pretty clearly says that there is only one.

Reply:
hmm, there are other `Drop` traits. for me "the" does not express that we are talking about the currently documented item. When reading your recommendation I thought that there may exist another `Drop` trait which is special, while this `Drop` trait is somehow not.
```rust
trait Drop { } // this is valid and not used in destructors
```